### PR TITLE
Synchronize pip and conda install methods

### LINKED ISSUE #40: Update contents of `make_env.yml` to match `requirements.txt`
The contents of the file `make_env.yml` seem to be outdated and out of synch with `requiremens.txt`:

`make_env.yml`
https://github.com/catalystneuro/buzsaki-lab-to-nwb/blob/7e7c9c2f7469c652b3de3f1d6bc2ffe06b24d426/make_env.yml#L1-L21
 
`requirements.txt`
https://github.com/catalystneuro/buzsaki-lab-to-nwb/blob/7e7c9c2f7469c652b3de3f1d6bc2ffe06b24d426/requirements.txt#L1-L19

Is there a reason for this mismatch?
I am intending to update the `make_env.yml` file to make it match as I have been using it in dandi to run our conversion. 

Can I move forward with this @CodyCBakerPhD ?

### PR DESCRIPTION
**Pull request by h-mayorquin**
Fixes #40 

# Description
This should allow the user to install the environment using conda. I also modified the `README.md ` file to indicate how to create the environment install with conda. I tested this on dandihub and it seems to be working. As you can see in the diff of `make_env.yml` file (conda) it now tracks the `requirements.txt` (pip).  That is, if we modify the `requirements.txt` file the conda environment will track those changes without further modifications. The only exception is the pandas library that had to be written explicitly in the conda file. There is some version compatibility issue that I could not identify but I hope I can solve after some discussion of the issues in the following section.

## Questions and discussion
1) As things stand we require a very specific snapshot of `nwb-conversiont-tools` in the `requirements.txt` file:
https://github.com/catalystneuro/buzsaki-lab-to-nwb/blob/8e139f8f617599a8c043ea32ce40ea6fd4493e05/requirements.txt#L17
Why is this? can we just install the latest available release or the latest available version? maybe we should document this choice?
2) The requirements of this repo and the ones of `nwb-conversion-tools` require sometimes the same package but not the same version. For example, we require `numpy==1.19.3` here but in `nwb-conversion-tools` we require `numpy>=1.21.0`.  I hightliht some of these differences in the code revision below.
3) Out of curiosity, why do we have python=3.7 for both this repo and `nwb-conversion-tools`, is there a package that does not work in newer versions or is it out of caution?

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,17 +1,16 @@
-
-pynwb==1.4.0
-tqdm==4.60.0
+pynwb>=1.4.0
```
Comment (h-mayorquin) [inline requirements.txt:1]: changed to aligin with `nwb-conversion-tools`

File: requirements.txt [hunk 2]
```diff
@@ -1,17 +1,16 @@
-
-pynwb==1.4.0
-tqdm==4.60.0
+pynwb>=1.4.0  
+tqdm>=4.60.0
 numpy==1.19.3
-pandas==1.2.3
-scipy==1.4.1
+pandas>=1.2.3
+scipy>=1.4.1
```
Comment (h-mayorquin) [inline requirements.txt:5]: changed to aligin with `nwb-conversion-tools`

File: requirements.txt [hunk 3]
```diff
@@ -1,17 +1,16 @@
...
 PyYAML==5.4
 jsonschema==3.2.0
 psutil==5.8.0
 lxml==4.6.5
 spikeextractors==0.9.7
-neo==0.9.0
+neo==0.9.0  # it is 0.9.0 in nwb-conversion-tools
 mat4py==0.5.0
 mat73==0.52
 nwb-conversion-tools @ git+https://github.com/catalystneuro/nwb-conversion-tools.git@55dc42853e3204b560f8e44b149ab0508e36852f
```
Comment (h-mayorquin) [inline requirements.txt:16]: Some specific state of the repo as far as I understand.

**Comment by CodyCBakerPhD** [comment]
> As things stand we require a very specific snapshot of nwb-conversiont-tools in the requirements.txt file:
buzsaki-lab-to-nwb/requirements.txt
> Line 17 in 8e139f8
>nwb-conversion-tools @ git+https://github.com/catalystneuro/nwb-conversion-tools.git@55dc42853e3204b560f8e44b149ab0508e36852f 
> Why is this? can we just install the latest available release or the latest available version? maybe we should document this choice?

For pinning nwb-conversion-tools versions we're probably just better off adding that inside each conversion project. I.e., `tingley_code_septal/requirements-tingley.txt` would have `nwb-conversion-tools==??.??.??` for whatever version or git hash was actually used on it. From experience, that is the only way to actually reproduce the same files in the same way months after the fact.


> The requirements of this repo and the ones of nwb-conversion-tools require sometimes the same package but not the same version. For example, we require numpy==1.19.3 here but in nwb-conversion-tools we require numpy>=1.21.0. I hightliht some of these differences in the code revision below.
I'm surprised there wasn't a security alert on that; we should be using `numpy >= 1.21.0` due to a recent security issue with buffer overflow. Not a huge deal tbh but still. This is why we use looser versions for minimal nwb-conversion-tools install these days, (using the `>=` instead of `==`), so a user can have a slightly different version if they want, and not have it be incompattible. But oftentimes there are minimal versions for basic operation.

**Comment by h-mayorquin** [comment]
@CodyCBakerPhD 
> For pinning nwb-conversion-tools versions we're probably just better off adding that inside each conversion project. I.e., `tingley_code_septal/requirements-tingley.txt` would have `nwb-conversion-tools==??.??.??` for whatever version or git hash was actually used on it. From experience, that is the only way to actually reproduce the same files in the same way months after the fact.

I think this makes sense to me. For every conversion we can have inside that directory the machinery to reproduce the libraries at the point when we did the conversion (so it would be both a requirements.txt and a conda.yml for each conversion directory). 

That said, what we should with the general `requirements.txt` (the one that I am modifying here)? should this just point out to the latest versio nof `nwb-conversion-tools` and `spikeextractors` and something that builds the environment for development? or what do you envision?
 
P.S. I modified the numpy version as well.

**Comment by CodyCBakerPhD** [comment]
General requirements for the package level would be fine. By that, I mean minimal loose version, i.e., `nwb-conversion-tools>=0.9.1` and that will install the most recent release by default.

**Comment by CodyCBakerPhD** [comment]
Also Python=3.7 is simply the minimal python requirement. `nwb-conversion-tools` supports 3.8 as well, and `hdmf`/`pynwb` support 3.9 and 3.10 I think

**Comment by h-mayorquin** [comment]
@CodyCBakerPhD 
> General requirements for the package level would be fine. By that, I mean minimal loose version, i.e., `nwb-conversion-tools>=0.9.1` and that will install the most recent release by default.

Then I suspect that the `requirements.txt` should only contain the following lines:

```
mat4py==0.5.0
mat73==0.52
hdf5storage>=0.1.18
nwb-conversion-tools>=0.9.1
spikeextractors>=0.9.7
```

All the rest of the lines are installed as a dependency from nwb-conversion-tools and are therefore redundant here. 

As you mentioned, at the end of every conversion we can recorder the pip freeze that we used for the conversion and add this the appropriate director. Maybe we can also write instructions in the readme.

**Comment by CodyCBakerPhD** [comment]
@h-mayorquin Sounds like a plan! Just let me know when this is ready for final review

**Comment by h-mayorquin** [comment]
@CodyCBakerPhD now it should be ready.
It builds in my sisten, is passing the installign tests here **but** fails in dandi for some reason. I think this is related to their build with scipy, should we be concerned about this?

**Comment by CodyCBakerPhD** [comment]
@h-mayorquin It's not a huge deal regardless; thanks for looking into this! Most important thing is the CI passes.

**Pull request merged**